### ISSUE TITLE
(bug): Nuxt typecheck is broken with nuxt/content@2.8.0

### ISSUE THREAD

**Issue by AdiRishi** [nuxt/content#2266]
### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.7.0
- CLI Version:      3.7.2
- Nitro Version:    2.6.2
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      modules
- Runtime Modules:  @nuxt/content@2.8.0
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-jvxdxz?file=README.md

### Describe the bug

With the release of `@nuxt/content@2.8.0` it seems like the typecheck command for Nuxt breaks.
Two main warnings/errors pop up
- One: the `@nuxtjs/mdc` package, specifically in the `ProseH1.vue` -> `ProseH6.vue` files have typescript issues that bubble up when running `nuxt typecheck`
- Two: `ProseCode.vue` seems to be resolved twice
(Check the logs section for the full error logs)

In my mind there are two problems here
1. The type issue that exists in the module which breaks typecheck for the entire project
2. The fact that Nuxt is checking types within node_modules in the first place. Surely this is not expected behaviour

### Logs

Warning logs when running `nuxt dev` or `nuxt typecheck`

```Shell
 WARN  [nuxt] Two component files resolving to the same name ProseCode:                                                                                                   11:48:06 am

 - /PATH_TO_REPO/node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseCode.vue
 - /PATH_TO_REPO/node_modules/.pnpm/@nuxt+content@2.8.0_vue@3.3.4/node_modules/@nuxt/content/dist/runtime/components/Prose/ProseCode.vue
```

Error logs when running `nuxt typecheck`
```Shell
node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH1.vue:19:45 - error TS18048: 'headings' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h1)
                                               ~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH1.vue:19:45 - error TS18048: 'headings.anchorLinks' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h1)
                                               ~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH2.vue:19:45 - error TS18048: 'headings' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h2)
                                               ~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH2.vue:19:45 - error TS18048: 'headings.anchorLinks' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h2)
                                               ~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH3.vue:19:45 - error TS18048: 'headings' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h3)
                                               ~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH3.vue:19:45 - error TS18048: 'headings.anchorLinks' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h3)
                                               ~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH4.vue:19:45 - error TS18048: 'headings' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h4)
                                               ~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH4.vue:19:45 - error TS18048: 'headings.anchorLinks' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h4)
                                               ~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH5.vue:19:45 - error TS18048: 'headings' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h5)
                                               ~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH5.vue:19:45 - error TS18048: 'headings.anchorLinks' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h5)
                                               ~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH6.vue:19:45 - error TS18048: 'headings' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h6)
                                               ~~~~~~~~

node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH6.vue:19:45 - error TS18048: 'headings.anchorLinks' is possibly 'undefined'.

19 const generate = computed(() => props.id && headings.anchorLinks.h6)
                                               ~~~~~~~~~~~~~~~~~~~~


Found 12 errors in 6 files.

Errors  Files
     2  node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH1.vue:19
     2  node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH2.vue:19
     2  node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH3.vue:19
     2  node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH4.vue:19
     2  node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH5.vue:19
     2  node_modules/.pnpm/@nuxtjs+mdc@0.1.2/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseH6.vue:19

 ERROR  Command failed with exit code 2: vue-tsc --noEmit                                                                                                                 11:48:08 am

  at makeError (node_modules/.pnpm/nuxi@3.7.2/node_modules/nuxi/dist/chunks/index3.mjs:625:11)
  at handlePromise (node_modules/.pnpm/nuxi@3.7.2/node_modules/nuxi/dist/chunks/index3.mjs:1688:26)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async Object.run (node_modules/.pnpm/nuxi@3.7.2/node_modules/nuxi/dist/chunks/typecheck.mjs:81:7)
  at async runCommand$1 (node_modules/.pnpm/nuxi@3.7.2/node_modules/nuxi/dist/shared/nuxi.c627b411.mjs:1644:5)
  at async runCommand$1 (node_modules/.pnpm/nuxi@3.7.2/node_modules/nuxi/dist/shared/nuxi.c627b411.mjs:1638:7)
  at async runMain$1 (node_modules/.pnpm/nuxi@3.7.2/node_modules/nuxi/dist/shared/nuxi.c627b411.mjs:1756:7)



 ERROR  Command failed with exit code 2: vue-tsc --noEmit                                                                                                                 11:48:08 am

error Command failed with exit code 1.
```

**Comment by xarthurx** [comment]
can confirm the issue.

**Comment by karngyan** [comment]
+1

**Comment by nobkd** [comment]
Might be fixed with https://github.com/nuxt-modules/mdc/commit/f65b3e95488fe6ef1c88d635f38bad0626624e37

**Comment by nobkd** [comment]
I can confirm that the type errors have been fixed with [v2.8.1](https://github.com/nuxt/content/releases/tag/v2.8.1).

---

The Prose override warnings have already been reported in #2262 and nuxt/nuxt#22876, and have been solved in Nuxt's edge channel with nuxt/nuxt#22882

**Comment by karngyan** [comment]
Thanks @nobkd  for the quick turnaround.

**Comment by nobkd** [comment]
Can you confirm that it works too?

**Comment by xarthurx** [comment]
@nobkd 

deleteing `node_modules` and `pnpm-lock` and do `pnpm i`, then `pnpm dev` still got this warning:

```
✔ Nuxt DevTools is enabled v0.8.2 (experimental)                                                             13:14:49

 WARN  [nuxt] Two component files resolving to the same name ProseCode:                                       13:14:49

 - C:/<projectFolder>/node_modules/.pnpm/@nuxtjs+mdc@0.1.3_rollup@2.79.1/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseCode.vue
 - C:/<projectFolder>/node_modules/.pnpm/@nuxt+content@2.8.1_rollup@2.79.1_vue@3.3.4/node_modules/@nuxt/content/dist/runtime/components/Prose/ProseCode.vue

```

**Comment by xarthurx** [comment]
Ah, sorry. I mis-read that the issue is fixed. 
I'll wait for the Edge-channel to be merged...

**Comment by adiramardiani** [comment]
![image](https://github.com/nuxt/content/assets/8703419/a22e9840-43ce-424a-971d-72e18e6edd0e)
the warning still here @nobkd @manniL 

is the patch has been merged ?

**Comment by karngyan** [comment]
On edge I can still see the warning:
```
❯ yarn dev
yarn run v1.22.19
$ nuxt dev
Nuxt 3.7.1-28223216.aa0ea8bf with Nitro 2.6.3-28222009.2a33f1f                                            4:50:10 PM
                                                                                                          4:50:11 PM
  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

ℹ Using Tailwind CSS from ~/assets/css/tailwind.css                                     nuxt:tailwindcss 4:50:11 PM
ℹ Tailwind Viewer: //_tailwind/                                                         nuxt:tailwindcss 4:50:12 PM

 WARN  [nuxt] Two component files resolving to the same name ProseCode:                                   4:50:12 PM

 - /.../ui/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProseCode.vue
 - /.../ui/node_modules/@nuxt/content/dist/runtime/components/Prose/ProseCode.vue


 WARN  [nuxt] Two component files resolving to the same name ProsePre:                                    4:50:12 PM

 - /.../ui/node_modules/@nuxtjs/mdc/dist/runtime/components/prose/ProsePre.vue
 - /.../ui/node_modules/@nuxt/content/dist/runtime/components/Prose/ProsePre.vue

ℹ ✨ optimized dependencies changed. reloading                                                           4:50:17 PM
ℹ ✨ new dependencies optimized: remark-reading-time                                                     4:50:18 PM
ℹ ✨ optimized dependencies changed. reloading                                                           4:50:18 PM
ℹ Vite client warmed up in 5308ms                                                                        4:50:18 PM
✔ Nitro built in 2718 ms                          
```

**Comment by adiramardiani** [comment]
for who using ci cd and restrict to warning message this is break my pipeline on live server, not good experience sorry

**Comment by manniL** [comment]
The ` WARN  [nuxt] Two component files resolving to the same name ProseCode:                                   4:50:12 PM` are resolved on nuxt edge via https://github.com/nuxt/nuxt/pull/22882.

**Comment by nobkd** [comment]
I also get it with edge-channel:
https://stackblitz.com/edit/github-fu8htb?file=package.json

Probably because both are modules and therefore have the same priority?
> The effect of this PR does not change behaviour; it only suppresses logs when overriding a lower priority component directory, to keep this within a patch release. We can consider bigger possible changes in the next minor.

**Comment by adiramardiani** [comment]
any update ?

**Comment by spyesx** [comment]
FYI – Got the same on a fresh install with:

```json
  "devDependencies": {
    "@nuxt/content": "^2.8.2",
    "nuxt": "^3.7.1"
  }
```

**Comment by Hime-Hina** [comment]
Was this bug fixed? I received the same warning.

![Snipaste_2023-09-08_21-13-35](https://github.com/nuxt/content/assets/17286494/09cf303a-9960-426c-89eb-b684829aa458)

**Comment by Celibioux** [comment]
same

**Comment by lumenghz** [comment]
Same issue, step to reproduce:

### Create fresh Nuxt Content Project

```shell
pnpm dlx nuxi@latest init content-app -t content
cd content-app
pnpm install --shamefully-hoist
```

### Just run

```shell
pnpm dev
```

### Output

<img width="1624" alt="image" src="https://github.com/nuxt/content/assets/3817355/1155de95-d038-4362-93df-356a03fb6ca9">

### Package version

- Nuxt: `3.7.1`
- Nuxt Content: `2.8.2`

**Comment by adiramardiani** [comment]
Is any update on this issue ?
Btw my development experience is so bad, I dont know this related to nuxt content, tailwind, or any other thing
But currently my startup time is CRAZY slow, any one experience the same thing ?

![image](https://github.com/nuxt/content/assets/8703419/ce558ddd-fb42-4b20-9890-d3bc890fd60b)

Just wait 3 minute for nitro built :(

**Comment by nobkd** [comment]
> But currently my startup time is CRAZY slow, any one experience the same thing ?

The slow startup sounds like a CLI problem:

https://github.com/nuxt/cli/issues/118
https://github.com/nuxt/cli/issues/136

To fix this in the meantime, you should be able to reset your `nuxi` to an earlier version e.g. `3.6.5`

**Comment by adiramardiani** [comment]
Hi All
Finally I can confirn ProseCode and ProsePre has gone after I update nuxt to 3.7.2
The issue of slow startup is fixed too

I hope this issue not comeback again

Any other one, is the issue fixed to on latest nuxt ?

**Comment by nobkd** [comment]
Can confirm this.
PR nuxt/nuxt#23156 should have fixed this.

**Comment by AdiRishi** [comment]
Yup can confirm that these issues are resolved. I'm happy to close this issue out unless anyone has any objections? Not sure what the usual procedure is, will maintainers close this or should I?

**Comment by nobkd** [comment]
No worries. Since this has been resolved, you can close the issue.

**Comment by manniL** [comment]
No, with Nuxt content they are still there, will be fixed by https://github.com/nuxt/content/pull/2302 though.

**Comment by nobkd** [comment]
@manniL
No, as far as I can see, it is solved: https://stackblitz.com/edit/github-fu8htb-vzqfsb?file=package.json
There are no warnings logged with `Content v2.8.2` and `Nuxt v3.7.2` anymore.

#2302 was one way to fix it, but the issue seems to be resolved with nuxt/nuxt#23156.
So #2302 doesn't seem to be needed anymore.

---

If you can share a reproduction where there are warnings with these package versions, then this issue should not be closed, of course.

**Comment by AdiRishi** [comment]
Alright, closing this issue now. I've confirmed the reported issues are no longer present as of `nuxt 3.7.3` and `@nuxt/content 2.8.2`
Thank you to everyone who helped fix this 😄
